src/nifler/emitter.nim

```diff
@@ -126,6 +108,45 @@ proc addEmpty*(em: var Emitter; count = 1) =
   for i in 1..count:
     em.emit ".", 1
 
+template addSuffixLit(em: var Emitter, suffix: string, body: typed) =
+  let suffixLit = '"' & suffix & '"'
+  var a = prepare(em, "suf")
+  em.addSep a
+  body
+  em.addSep a
+  em.emit suffixLit, suffixLit.len
+  em.patch(a)
+
+template addSuffixLitDispatch(em: var Emitter, suffix: string, body: typed) =
+  if suffix.len > 0:
+    addSuffixLit(em, suffix):
+      body
+  else:
+    body
+
+
+proc addStrLit*(em: var Emitter; s: string) =
+  var r = "\""
+  var l = em.lineLen + 1
+  var lastPart = 1
+  var afterNewline = false
+  for c in s:
+    if c in ControlChars:
+      r.escape c
+      inc l, 3
+      inc lastPart, 3
+    else:
+      r.add c
+      inc l, 1
+      inc lastPart, 1
+    afterNewline = false
+  r.add "\""
+  em.emit r, lastPart
+
+proc addStrLit*(em: var Emitter; s, suffix: string) =
```

Comment:
No need for this API.

Reply:
The old one provides this API
```nim
proc addStrLit*(em: var Emitter; s, suffix: string)
```
Shouldn't strings have suffix?